typespec-extension/src/code-model-builder.ts

```diff
@@ -706,22 +704,20 @@ export class CodeModelBuilder {
     return Boolean(this.options["advanced-versioning"]);
   }
 
-  private getOperationExample(operation: Operation): any | undefined {
-    if (operation.projectionSource?.kind === "Operation") {
-      // always use the projectionSource, if available
-      operation = operation.projectionSource;
-    }
-    let operationExample = this.operationExamples.get(operation);
-    if (!operationExample && operation.sourceOperation) {
-      // if the operation is customized in client.tsp, the operation would be different from that of main.tsp
-      // try the operation.sourceOperation
-      operation = operation.sourceOperation;
-      if (operation.projectionSource?.kind === "Operation") {
-        operation = operation.projectionSource;
+  private getOperationExample(operation: HttpOperation): Record<string, any> | undefined {
+    const httpOperationExamples = getHttpOperationExamples(this.sdkContext, operation);
+    if (httpOperationExamples && httpOperationExamples.length > 0) {
+      const operationExamples: Record<string, any> = {};
+      for (const example of httpOperationExamples) {
+        const operationExample = example.rawExample;
+        operationExample["x-ms-original-file"] = pathToFileURL(example.filePath).toString();
+        operationExamples[operationExample.title ?? operationExample.operationId ?? operation.operation.name] =
```

Comment:
maybe the `operation.operation.name` can be removed, as TCGC will ensure either `title` or `operationId`:
https://github.com/Azure/typespec-azure/pull/1076/files#diff-e186052723c0e3dfe654f3a7fd0f36bd8126c371ad017fdee5a8643ba3d8da10R87

Reply:
The rawExample is type `any`, so we don't have any constraint on type of this behavior.

Though I am actually relying on their logic on duplicate title.
https://github.com/Azure/typespec-azure/pull/1076/files#diff-d276532a6e7d2da177119679a7dfdc92a59980b7bbb70000f99ad0e4d50f3183R180